### PR TITLE
Improved the ios installer

### PR DESCRIPTION
**Pull request by lucamaraschi**
Improved the installation on ios, making the downloading of the zip package deterministic.
Overall refactor of the control flow and added dependencies to the `package.json` to ease the execution.

### CODE REVIEW

**Comment by dzhuowen** [comment]
Thanks for the PR. I tried using it but the script did finish. Look like you didn't use fastfall correctly.

File: bin/ios_setup.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    myProj.addFramework('./ios/Frameworks/FBSDKCoreKit.framework', { 'customFramework': true, 'target': target, 'link': true })
+    myProj.addFramework('./ios/Frameworks/FBSDKShareKit.framework', { 'customFramework': true, 'target': target, 'link': true })
+    myProj.addFramework('./ios/Frameworks/FBSDKLoginKit.framework', { 'customFramework': true, 'target': target, 'link': true })
+
+    // WARNING: this will overwrite any existing search paths
+    myProj.updateBuildProperty('FRAMEWORK_SEARCH_PATHS', '"$(PROJECT_DIR)/Frameworks/"')
+    fs.writeFileSync(myProjPath, myProj.writeSync())
+    console.log('Finished updating ' + myProjPath)
+
+    next(files)
```
Comment (dzhuowen) [inline bin/ios_setup.js]: `next(null, files)`

File: bin/ios_setup.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  })
+}
+
+function updatePlist (next, files) {
```
Comment (dzhuowen) [inline bin/ios_setup.js]: function updatePlist (files, next) {

**Comment by lucamaraschi** [comment]
@dzhuowen fixed!

**Comment by lucamaraschi** [comment]
Any feedback?

**Comment by delvedor** [comment]
Hi! Any chance to see this landed? :)

**Comment by lucamaraschi** [comment]
@dzhuowen ping

**Comment by dzhuowen** [comment]
@lucamaraschi  Thanks for the PR!

**Pull request merged**